README.md

```diff
@@ -21,59 +21,145 @@ locale. The "core" `Faker\Generator` contains a set of predefined of methods; `f
 `address()`, `companyName()` etc, which is just syntactic sugar for calling the
 extensions.
 
-Here is how to use an extension:
+### Registering an extension
+
+We use PSR-11 to configure extensions:
 
 ```php
-$faker->ext(Foo::class)->bar();
+<?php
+
+use Example\Person;
+use Faker\ContainerBuilder;
+use Faker\Factory;
+use Psr\Container;
+
+$containerBuilder = new ContainerBuilder();
+
+$containerBuilder->register(Person::class);
+
+/** @var Container\ContainerInterface $container */
+$container = $containerBuilder->build();
+
+$faker = Factory::create(
+    null, 
+    $container
+);
 ```
 
-Example implementation of core methods:
+### Implementing an extension
+
+A empty marker interface is provided:
 
 ```php
-public function firstName() {
-    return $faker->ext(PersonInterface::class)->firstName();
+<?php
+
+namespace Faker\Extension;
+
+interface Extension
+{
 }
 ```
 
-We use PSR-11 to configure extensions:
+This marker interface can be implemented to help with auto-disovery of extensions:
 
 ```php
-use Faker\Factory;
+<?php
 
-$builder = new ContainerBuilder();
-$builder->register(Foo:class);
-$psr11Container = $builder->build();
+namespace Example;
 
-$faker = Factory::create(null, $psr11Container);
+use Faker\Extension;
+
+final class Person implements Extension\Extension
```

Comment:
```suggestion
final class Person implements Faker\Extension
```

Reply:
`Faker\Extension` is imported here, and I figured it would make sense to have a `Faker\Extension` namespace, but we could of course have

- `Faker\Extension` as interface
- `Faker\GeneratorAwareExtension` as interface

What do you prefer?

Reply:
Oops, I didnt see that. 
For all the "core" extensions, we also need interfaces. I agree that it makes sense to have `Faker\Extension\PersonExtension`. 

Both `Faker\Extension\PersonExtension` and `Faker\Extension\GeneratorAwareExtension` should extend `Faker\Extension\Extension`. 

:+1: for an extension interface.